projects/hslayers/src/common/layers/interpolated-source.ts

```diff
@@ -0,0 +1,190 @@
+import IDW from 'ol-ext/source/IDW';
+
+import VectorSource from 'ol/source/Vector';
+import {Feature} from 'ol';
+import {Fill, Stroke, Style, Text} from 'ol/style';
+import {GeoJSON} from 'ol/format';
+import {Geometry} from 'ol/geom';
+import {Projection} from 'ol/proj';
+import {Subject} from 'rxjs';
+import {Vector as VectorLayer} from 'ol/layer';
+import {containsExtent, equals} from 'ol/extent';
+
+export const NORMALIZED_WEIGHT_PROPERTY_NAME = 'hs_normalized_IDW_value';
+
+export interface InterpolatedSourceOptions {
+  features?: Feature<Geometry>[];
+  weight?: string;
+  loader?(params: any): Promise<Feature[]>;
+  /**
+   * Set custom colors for Interpolated source
+   * @param value - Representing value
+   * @param data - RGBA array
+   * @param i - index in the RGBA array
+   */
+  customColors?: {
+    value: number;
+    data: Uint8ClampedArray;
+    i: number;
+  };
+}
+
+export class InterpolatedSource extends IDW {
+  cancelUrlRequest: Subject<void> = new Subject();
+  constructor(private options: InterpolatedSourceOptions) {
+    super({
+      // Source that contains the data
+      source: new VectorSource({
+        features: options.features,
```

Comment:
Need to normalize those too

Reply:
Do we want to keep those initial features all the time?

Reply:
no, just until loader is called if it's provided